packages/opentelemetry-plugin-mongodb-core/src/mongodb.ts

```diff
@@ -0,0 +1,236 @@
+        options: {} | Function,
+        callback: Func<unknown>
+      ): mongodb.Server {
+        const currentSpan = plugin._tracer.getCurrentSpan();
+        // @ts-ignore
+        const resultHandler =
+          typeof options === 'function' ? options : callback;
+        if (
+          currentSpan === null ||
+          typeof resultHandler !== 'function' ||
+          typeof commands !== 'object'
+        ) {
+          return original.apply(this, (arguments as unknown) as unknown[]);
+        }
+        const command = commands instanceof Array ? commands[0] : commands;
+        const type = plugin._getCommandType(command, operationName);
+        const span = plugin._tracer.startSpan(`mongodb.${type}`, {
+          parent: currentSpan,
+          kind: SpanKind.CLIENT,
+        });
+        plugin._populateAttributes(
+          span,
+          ns,
+          command,
+          this as MongoInternalTopology
+        );
+        if (typeof options === 'function') {
+          return original.call(
+            this,
+            ns,
+            commands,
+            plugin._patchEnd(span, options as Func<unknown>)
+          );
+        } else {
+          return original.call(
+            this,
+            ns,
+            commands,
+            options,
+            plugin._patchEnd(span, callback)
+          );
+        }
+      };
+    };
+  }
+
+  /**
+   * Get the mongodb command type from the object.
+   * @param command Internal mongodb command object
+   * @param defaulType the default type to return if we could not find a
+   *  specific command.
+   */
+  private _getCommandType(command: MongoInternalCommand, defaulType: string) {
+    if (command.createIndexes !== undefined) {
+      return 'createIndexes';
+    } else if (command.findandmodify !== undefined) {
+      return 'findAndModify';
+    } else if (command.ismaster !== undefined) {
+      return 'isMaster';
+    } else if (command.count !== undefined) {
+      return 'count';
+    } else {
+      return defaulType;
+    }
+  }
+
+  /**
+   * Populate span's attributes by fetching related metadata from the context
+   * @param span span to add attributes to
+   * @param ns mongodb namespace
+   * @param command mongodb internal representation of a command
+   * @param topology mongodb internal representation of the network topology
+   */
+  private _populateAttributes(
+    span: Span,
+    ns: string,
+    command: MongoInternalCommand,
+    topology: MongoInternalTopology
+  ) {
+    // add network attributes to determine the remote server
+    if (topology && topology.s && topology.s.options) {
+      span.setAttribute('peer.hostname', `${topology.s.options.host}`);
+      span.setAttribute('peer.port', `${topology.s.options.port}`);
+    }
+    // add database related attributes
+    span.setAttribute('db.instance', `${ns}`);
+    span.setAttribute('db.type', `mongodb`);
+    span.setAttribute('component', 'mongodb-core');
```

Comment:
I'd suggest to use `.setAttributes()` instead of `.setAttribute()` to sets attributes to the span.
```
span.setAttributes({
'db.instance': `${ns}`,
'db.type': `mongodb`,
// ... and other attributes
});
```
May be we can (re)use [AttributeNames](https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-plugin-postgres/src/enums.ts#L17) enum.